Source/DafnyCore/Rewriters/UselessOldAndRustAssertLinter.cs

```diff
@@ -46,5 +46,18 @@ class UselessOldLinterVisitor : TopDownVisitor<Unit> {
       }
       return base.VisitOneExpr(expr, ref st);
     }
+
+    protected override bool VisitOneStmt(Statement stmt, ref Unit st) {
+      // Check if the statement has the three tokens "assert", "!", and "(" next to each other. If so, that may
+      // be a mistake, especially if the programmer is coming from Rust.
+      if (stmt is AssertStmt { Expr: UnaryOpExpr { Op: UnaryOpExpr.Opcode.Not, E: ParensExpression } negationExpression }) {
+        if (stmt.tok.pos + 6 == negationExpression.tok.pos) {
+          reporter.Warning(MessageSource.Rewriter, ErrorDetail.ErrorID.None, stmt.tok,
+            "You have written an assert statement with a negated condition, but the lack of whitespace between 'assert' and '!' " +
+            "suggests you may be used to Rust and have accidentally negated the asserted condition. Consider adding a space between these two.");
```

Comment:
```suggestion
            "suggests you may be used to Rust and have accidentally negated the asserted condition. If that is the case, please remove the "!" and the parentheses or, if it was intended, please add a space between these two.");
```
Otherwise, the "consider" is not in line with the previous statement.